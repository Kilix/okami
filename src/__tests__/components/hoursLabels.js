import React from 'react'
import {mount} from 'enzyme'
import toJson from 'enzyme-to-json'
import frLocale from 'date-fns/locale/fr'
import mockdate from 'mockdate'

import HoursLabels from '../../components/hoursLabels'

describe('HoursLabels', () => {
  beforeEach(() => {
    mockdate.set('3/3/2017', 0)
  })
  afterEach(() => {
    mockdate.reset()
  })

  describe('daily', () => {
    const ctx = {
      type: 'daily',
      locale: frLocale,
      hourFormat: 'HH',
      showWeekend: true,
      rowHeight: 30,
      endHour: 'PT22H',
      startHour: 'PT06H',
      currentDay: new Date(2017, 9, 10, 0, 0, 0, 0),
    }
    test('render', () => {
      const tree = mount(<HoursLabels />, {context: ctx})
      expect(toJson(tree)).toMatchSnapshot()
    })
    test('render Child', () => {
      const tree = mount(<HoursLabels renderChild={({idx, ...props}) => <span {...props} />} />, {
        context: ctx,
      })
      expect(toJson(tree)).toMatchSnapshot()
    })
    test('function as a Child', () => {
      const tree = mount(
        <HoursLabels>
          {({hours}) => <div>{hours.map((w, idx) => <span key={idx}>{w}</span>)}</div>}
        </HoursLabels>,
        {context: ctx}
      )
      expect(toJson(tree)).toMatchSnapshot()
    })
    test('render with no weekend', () => {
      const tree = mount(<HoursLabels />, {context: {...ctx, showWeekend: false}})
      expect(toJson(tree)).toMatchSnapshot()
    })
  })
  describe('weekly', () => {
    const ctx = {
      type: 'weekly',
      locale: frLocale,
      hourFormat: 'HH',
      showWeekend: true,
      rowHeight: 30,
      endHour: 'PT22H',
      startHour: 'PT06H',
      startWeek: new Date(2017, 9, 10, 0, 0, 0, 0),
      offset: 1,
    }
    test('render', () => {
      const tree = mount(<HoursLabels />, {context: ctx})
      expect(toJson(tree)).toMatchSnapshot()
    })
    test('render with style', () => {
      const tree = mount(<HoursLabels style={{color: 'orange'}} />, {context: ctx})
      expect(toJson(tree)).toMatchSnapshot()
    })
    test('render Child', () => {
      const tree = mount(<HoursLabels renderChild={({idx, ...props}) => <span {...props} />} />, {
        context: ctx,
      })
      expect(toJson(tree)).toMatchSnapshot()
    })
    test('function as a Child', () => {
      const tree = mount(
        <HoursLabels>
          {({hours}) => <div>{hours.map((w, idx) => <span key={idx}>{w}</span>)}</div>}
        </HoursLabels>,
        {context: ctx}
      )
      expect(toJson(tree)).toMatchSnapshot()
    })
    test('render with no weekend', () => {
      const tree = mount(<HoursLabels />, {context: {...ctx, showWeekend: false}})
      expect(toJson(tree)).toMatchSnapshot()
    })
  })
})
