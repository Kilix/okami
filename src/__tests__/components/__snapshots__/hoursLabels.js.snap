// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HoursLabels daily function as a Child 1`] = `"<div><span>04</span><span>05</span><span>06</span><span>07</span><span>08</span><span>09</span><span>10</span><span>11</span><span>12</span><span>13</span><span>14</span><span>15</span><span>16</span><span>17</span><span>18</span><span>19</span></div>"`;

exports[`HoursLabels daily render 1`] = `"<div style=\\"padding-top: 0px;\\"><div style=\\"height: 30px;\\">04</div><div style=\\"height: 30px;\\">05</div><div style=\\"height: 30px;\\">06</div><div style=\\"height: 30px;\\">07</div><div style=\\"height: 30px;\\">08</div><div style=\\"height: 30px;\\">09</div><div style=\\"height: 30px;\\">10</div><div style=\\"height: 30px;\\">11</div><div style=\\"height: 30px;\\">12</div><div style=\\"height: 30px;\\">13</div><div style=\\"height: 30px;\\">14</div><div style=\\"height: 30px;\\">15</div><div style=\\"height: 30px;\\">16</div><div style=\\"height: 30px;\\">17</div><div style=\\"height: 30px;\\">18</div><div style=\\"height: 30px;\\">19</div></div>"`;

exports[`HoursLabels daily render Child 1`] = `"<div style=\\"padding-top: 0px;\\"><span>04</span><span>05</span><span>06</span><span>07</span><span>08</span><span>09</span><span>10</span><span>11</span><span>12</span><span>13</span><span>14</span><span>15</span><span>16</span><span>17</span><span>18</span><span>19</span></div>"`;

exports[`HoursLabels daily render with no weekend 1`] = `"<div style=\\"padding-top: 0px;\\"><div style=\\"height: 30px;\\">04</div><div style=\\"height: 30px;\\">05</div><div style=\\"height: 30px;\\">06</div><div style=\\"height: 30px;\\">07</div><div style=\\"height: 30px;\\">08</div><div style=\\"height: 30px;\\">09</div><div style=\\"height: 30px;\\">10</div><div style=\\"height: 30px;\\">11</div><div style=\\"height: 30px;\\">12</div><div style=\\"height: 30px;\\">13</div><div style=\\"height: 30px;\\">14</div><div style=\\"height: 30px;\\">15</div><div style=\\"height: 30px;\\">16</div><div style=\\"height: 30px;\\">17</div><div style=\\"height: 30px;\\">18</div><div style=\\"height: 30px;\\">19</div></div>"`;

exports[`HoursLabels weekly function as a Child 1`] = `"<div><span>04</span><span>05</span><span>06</span><span>07</span><span>08</span><span>09</span><span>10</span><span>11</span><span>12</span><span>13</span><span>14</span><span>15</span><span>16</span><span>17</span><span>18</span><span>19</span></div>"`;

exports[`HoursLabels weekly render 1`] = `"<div style=\\"padding-top: 30px;\\"><div style=\\"height: 30px;\\">04</div><div style=\\"height: 30px;\\">05</div><div style=\\"height: 30px;\\">06</div><div style=\\"height: 30px;\\">07</div><div style=\\"height: 30px;\\">08</div><div style=\\"height: 30px;\\">09</div><div style=\\"height: 30px;\\">10</div><div style=\\"height: 30px;\\">11</div><div style=\\"height: 30px;\\">12</div><div style=\\"height: 30px;\\">13</div><div style=\\"height: 30px;\\">14</div><div style=\\"height: 30px;\\">15</div><div style=\\"height: 30px;\\">16</div><div style=\\"height: 30px;\\">17</div><div style=\\"height: 30px;\\">18</div><div style=\\"height: 30px;\\">19</div></div>"`;

exports[`HoursLabels weekly render Child 1`] = `"<div style=\\"padding-top: 30px;\\"><span>04</span><span>05</span><span>06</span><span>07</span><span>08</span><span>09</span><span>10</span><span>11</span><span>12</span><span>13</span><span>14</span><span>15</span><span>16</span><span>17</span><span>18</span><span>19</span></div>"`;

exports[`HoursLabels weekly render with no weekend 1`] = `"<div style=\\"padding-top: 30px;\\"><div style=\\"height: 30px;\\">04</div><div style=\\"height: 30px;\\">05</div><div style=\\"height: 30px;\\">06</div><div style=\\"height: 30px;\\">07</div><div style=\\"height: 30px;\\">08</div><div style=\\"height: 30px;\\">09</div><div style=\\"height: 30px;\\">10</div><div style=\\"height: 30px;\\">11</div><div style=\\"height: 30px;\\">12</div><div style=\\"height: 30px;\\">13</div><div style=\\"height: 30px;\\">14</div><div style=\\"height: 30px;\\">15</div><div style=\\"height: 30px;\\">16</div><div style=\\"height: 30px;\\">17</div><div style=\\"height: 30px;\\">18</div><div style=\\"height: 30px;\\">19</div></div>"`;

exports[`HoursLabels weekly render with style 1`] = `"<div style=\\"color: orange; padding-top: 30px;\\"><div style=\\"height: 30px;\\">04</div><div style=\\"height: 30px;\\">05</div><div style=\\"height: 30px;\\">06</div><div style=\\"height: 30px;\\">07</div><div style=\\"height: 30px;\\">08</div><div style=\\"height: 30px;\\">09</div><div style=\\"height: 30px;\\">10</div><div style=\\"height: 30px;\\">11</div><div style=\\"height: 30px;\\">12</div><div style=\\"height: 30px;\\">13</div><div style=\\"height: 30px;\\">14</div><div style=\\"height: 30px;\\">15</div><div style=\\"height: 30px;\\">16</div><div style=\\"height: 30px;\\">17</div><div style=\\"height: 30px;\\">18</div><div style=\\"height: 30px;\\">19</div></div>"`;
